Automattic/woocommerce-payments
# Add country-specific test card numbers for credit card testing

Closes https://github.com/Automattic/woocommerce-payments/issues/9619

#### Changes proposed in this Pull Request
This PR adds support for country-specific test card numbers in test mode. When a merchant is testing payments, they will now see the appropriate test card number based on their account's country. The test card numbers are sourced from [Stripe's documentation](https://docs.stripe.com/testing?testing-method=card-numbers#international-cards).








#### Testing instructions





* check whether the array added in `Country_Test_Cards` is in line with countries list mentioned in [Stripe docs](https://docs.stripe.com/testing?testing-method=card-numbers#international-cards)
* you can test with different account countries by adding something like `$account_response['country'] = 'GB';` before [this line](https://github.com/Automattic/transact-platform-server/blob/trunk/server/wp-content/rest-api-plugins/endpoints/wcpay/class-accounts-controller.php#L969) on the server
* confirm that for different account countries, different card numbers are being returned
* confirm that for unknown account country, the default `4242 ... ` card number is used
* enable other payment methods and confirm that when they are displayed on the checkout page, no errors/exceptions occur
* (optional, nice to have) feel free to create a couple of JN sites with different account countries and see the effect of this PR there
* if you checked only one type of checkout (e.g. only Blocks or only shortcode) please repeat the smoke tests for another checkout type as well

-------------------


**Post merge**
